support-e2e/tests/utils/testUserDetails.ts

```diff
@@ -10,3 +11,64 @@ export const setTestUserDetails = async (
 	await page.getByLabel('First name').fill(firstName);
 	await page.getByLabel('Last name').fill(lastName);
 };
+
+export const setTestUserDetails = async (
+	page: Page,
+	testDetails: TestDetails,
+) => {
+	await setTestUserRequiredDetails(
+		page,
+		testDetails.fields.firstName,
+		testDetails.fields.lastName,
+		testDetails.fields.email,
+	);
+
+	if (testDetails.fields.addresses && testDetails.fields.addresses.length > 1) {
+		await page
+			.getByRole('checkbox', {
+				name: 'Billing address same as delivery address',
+			})
+			.uncheck();
+	}
+
+	if (testDetails.fields.addresses) {
+		let index = 0;
+		// To run in sequence using async/await, for required over forEach
+		for (const address of testDetails.fields.addresses) {
```

Comment:
this is clever - but maybe better to just make it really simple and repeat ourselves? We're only going to have two addresses ever right? Delivery and billing

Reply:
Yes, its not perfect, but just wanted to keep repeat code to a minimum thus using the loop. Hope that's ok.